solidity/contracts/isms/routing/timelock.md

```diff
@@ -0,0 +1,241 @@
+}}%%
+
+graph TB
+    MB[Mailbox]
+
+    OuterAgg[Outer AggregationISM<br/>1-of-2: EITHER path]
+    style OuterAgg fill:purple
+
+    InnerAgg[Inner AggregationISM<br/>2-of-2: BOTH required]
+    style InnerAgg fill:purple
+
+    PISM[PausableISM<br/>Watcher Control]
+    style PISM fill:red
+
+    TL[TimelockRouter<br/>Time Delay]
+    style TL fill:orange
+
+    ZK[ZkProofISM<br/>Rollup Finality]
+    style ZK fill:blue
+
+    Watcher([Watcher])
+    style Watcher fill:black
+
+    MB -- "verify()" --> OuterAgg
+    OuterAgg -- "Path 1:<br/>Fast Optimistic" --> InnerAgg
+    OuterAgg -- "Path 2:<br/>Slow Finality" --> ZK
+
+    InnerAgg -- "verify()" --> PISM
+    InnerAgg -- "verify()" --> TL
+
+    PISM -- "!paused?" --> InnerAgg
+    TL -- "readyAt <= now?" --> InnerAgg
+    InnerAgg -- "BOTH passed<br/>✓ fast path" --> OuterAgg
+    ZK -- "valid proof?<br/>✓ slow path" --> OuterAgg
+    OuterAgg -- "EITHER passed<br/>✓ process" --> MB
+
+    Watcher -. "pause()" .-> PISM
+```
+
+**Verification flow:**
+
+**Path 1 (Fast Optimistic)**: Inner 2-of-2 aggregation requires BOTH:
+
+- `PausableISM.verify()` passes (not paused by watcher)
+- `TimelockRouter.verify()` passes (timelock expired)
+- **Result**: Message delivered after timelock window, unless paused
+
+**Path 2 (Slow Finality)**: Bypass optimistic layer entirely:
+
+- `ZkProofISM.verify()` passes (valid ZK proof or cryptographic finality)
+- **Result**: Message delivered immediately with cryptographic proof
+
+**Outer 1-of-2 aggregation**: Message succeeds if EITHER path passes
+
+**Benefits:**
+
+- **Normal case**: Fast optimistic path (low cost, ~1 hour latency)
+- **Emergency case**: Slow finality path bypasses paused optimistic layer
+- **Redundancy**: Two independent security mechanisms
+
+**Watcher workflow:**
+
+1. **Monitor**: Watch for `MessageQueued` events from TimelockRouter
+2. **Validate**: Check if message is valid (verify proofs, check state consistency)
+3. **Pause if fraud detected**: Call `pausableIsm.pause()` to halt message processing
+4. **Investigate & resolve**: Fix issues, then call `pausableIsm.unpause()`
+
+**Security comparison:**
+
+| Security Model                             | Time Delay | Fraud Prevention    | Cryptographic Proof | Redundancy      | Security Level |
+| ------------------------------------------ | ---------- | ------------------- | ------------------- | --------------- | -------------- |
+| TimelockRouter only                        | ✓          | ✗                   | ✗                   | ✗ Single path   | Low            |
+| 2-of-2 Agg (Pausable + Timelock)           | ✓          | ✓ Watcher pause     | ✗                   | ✗ Single path   | Medium         |
+| ZkProofISM only                            | ✗          | N/A                 | ✓                   | ✗ Single path   | Medium         |
+| **1-of-2 Dual-Path (Optimistic OR Proof)** | **✓**      | **✓ Watcher pause** | **✓**               | **✓ Two paths** | **Maximum**    |
```

Comment:
I don't see how each individual ism has a security level of "medium" but a 1-of-2 agg ism has a security level of "maximum"